OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -82,6 +95,11 @@ public sealed class HackyAIInfo : IBotInfo, ITraitInfo
 		[Desc("Radius in cells around the center of the base to expand.")]
 		public readonly int MaxBaseRadius = 20;
 
+		[Desc("Radius in cells around each building with ProvideBuildableArea",
+		"to check for a 3x3 area of water where naval structures can be built.",
```

Comment:
I'm not certain if I like the sound of this. RA2 naval yards are definitely bigger than 3x3 and IIRC that GDI navalyard c'n'p for TS is also 4x3. Can't you customize this that instead of a fixed size, it checks for the biggest foundation of the buildings listed in NavalProduction?